src/gc.rs

```diff
@@ -0,0 +1,70 @@
+/// Applications can register callback functions which will be called before and
+/// after certain garbage collection operations.  Allocations are not allowed in
+/// the callback functions, you therefore cannot manipulate objects (set or
+/// delete properties for example) since it is possible such operations will
+/// result in the allocation of objects.
+#[repr(C)]
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+pub struct GCType(u32);
+
+pub const TYPE_SCAVENGE: GCType = GCType(1);
+
+pub const MINOR_MARK_COMPACT: GCType = GCType(2);
+
+pub const MARK_SWEEP_COMPACT: GCType = GCType(4);
+
+pub const INCREMENTAL_MARKING: GCType = GCType(8);
+
+pub const PROCESS_WEAK_CALLBACK: GCType = GCType(16);
+
+pub const TYPE_ALL: GCType = GCType(31);
```

Comment:
Use eum

Reply:
If I do enum then we won't be able to do bitor on them

Reply:
then prefix each with `GC_TYPE_` and add a test demoing or-ing them.

Reply:
or maybe use rust namespace 
```rust
mod gc_type {
   pub const TYPE_SCAVENGE: GCType = GCType(1);
  // ...
}
```